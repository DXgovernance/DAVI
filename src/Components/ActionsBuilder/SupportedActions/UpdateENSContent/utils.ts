import contenthash from 'content-hash';
import { MAINNET_ID, LOCALHOST_ID, GOERLI_ID } from 'utils';
import { utils } from 'ethers';

const DEFAULT_NAMEHASH =
  '0x0000000000000000000000000000000000000000000000000000000000000000';

export const convertToNameHash = (name: string) => {
  if (!name) {
    return DEFAULT_NAMEHASH;
  }
  return utils.namehash(name);
};

export const convertToContentHash = (ipfsHash: string) => {
  if (!ipfsHash) {
    return '';
  }
  return contenthash.fromIpfs(ipfsHash);
};

export const convertToIpfsHash = (contentHash: string) => {
  if (!contentHash) return null;
  return contenthash.decode(contentHash);
};

export const getIpfsUrl = (ipfsHash: string) => {
  const ipfsRoot = 'ipfs://';
  return `${ipfsRoot}${ipfsHash}`;
};

export const isValidChainId = (chainId: number) => {
  if (!chainId || chainId === LOCALHOST_ID) {
    return MAINNET_ID;
  } else {
    return chainId;
  }
};

export const isAvailableOnENS = (chainId: number) => {
  const validChainId = isValidChainId(chainId);
  const ensNetworks = [MAINNET_ID, GOERLI_ID];
  return ensNetworks.includes(validChainId);
};
