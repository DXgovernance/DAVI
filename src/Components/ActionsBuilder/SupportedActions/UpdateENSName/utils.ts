import namehash from 'eth-ens-namehash';
import contenthash from 'content-hash';
import { useState, useEffect } from 'react';

export const convertToNameHash = (name: string) => namehash.hash(name);

export const convertToContentHash = (ipfsHash: string) =>
  contenthash.fromIpfs(ipfsHash);

export const convertToIPFSHash = (contentHash: string) => {
  if (!contentHash) return null;
  return contenthash.decode(contentHash);
};

export const useDebounce = (value: string, delay: number) => {
  const [debouncedValue, setDebouncedValue] = useState(value);

  useEffect(() => {
    const handler = setTimeout(() => {
      setDebouncedValue(value);
    }, delay);

    return () => {
      clearTimeout(handler);
    };
  }, [value, delay]);

  return debouncedValue;
};
