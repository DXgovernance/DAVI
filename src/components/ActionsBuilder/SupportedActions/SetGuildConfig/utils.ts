import { FIELDS } from './constants';
import { BigNumber } from 'ethers';
import { SetGuildConfigFields, ControlField } from './types';

export const bn = (n: string | number | BigNumber) => BigNumber.from(n);

type UpdatedValuesReturn = {
  [key in ControlField]: {
    newValue: BigNumber;
    currentValue: BigNumber;
  };
};
export const getUpdatedValues = (
  current: SetGuildConfigFields,
  modifyed: SetGuildConfigFields
): UpdatedValuesReturn | {} => {
  if (!current || !modifyed) return {};
  return Object.keys(current).reduce((acc, key) => {
    if (!FIELDS.map(f => f.name).includes(key)) return acc;
    const currentValue = bn(current[key] || 0);
    const newValue = bn(modifyed[key] || 0);

    if (!currentValue.eq(newValue)) {
      return {
        ...acc,
        [key]: { newValue, currentValue },
      };
    }
    return acc;
  }, {});
};

export const pickUpdatedOrDefaultValue = (
  current: BigNumber,
  modifyed: BigNumber
): BigNumber => {
  if (BigNumber.isBigNumber(current)) {
    if (BigNumber.isBigNumber(modifyed) && !bn(modifyed).eq(current)) {
      return bn(modifyed);
    }
    return bn(current);
  } else if (BigNumber.isBigNumber(modifyed)) {
    return bn(modifyed);
  }
  return bn(0);
};
