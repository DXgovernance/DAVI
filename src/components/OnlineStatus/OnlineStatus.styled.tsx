import styled from 'styled-components';

export const OnlineStatusWrapper = styled.div`
  width: fit-content;
  margin: 0 auto;
  border: ${({ theme }) => `1px solid ${theme.colors.border1}`};
  padding: 40px;
  border-radius: 4px;
`;

export const Title = styled.h3`
  margin-bottom: 0;
  font-size: 1.5rem;
`;
export const SubTitle = styled.p`
  font-size: 1rem;
`;
