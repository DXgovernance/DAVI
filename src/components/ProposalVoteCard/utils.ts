import { ConfirmVoteProposalProps, VoteOnProposalProps } from './types';

export const voteOnProposal = ({
  votingPowerAtProposalSnapshot,
  votingPowerAtProposalCurrentSnapshot,
}: VoteOnProposalProps) => {
  const hasNoVotingPower =
    votingPowerAtProposalSnapshot &&
    Number(votingPowerAtProposalSnapshot?.toString()) <= 0;

  const hasVotingPowerAtCurrentSnapshot =
    votingPowerAtProposalCurrentSnapshot &&
    Number(votingPowerAtProposalCurrentSnapshot?.toString()) > 0;
  return {
    hasNoVotingPower,
    hasVotingPowerAtCurrentSnapshot,
  };
};

export const confirmVoteProposal = ({
  proposal,
  contract,
  selectedOption,
  userVotingPower,
  createTransaction,
  cb,
}: ConfirmVoteProposalProps) => {
  createTransaction(
    `Vote on proposal ${proposal?.title}`,
    async () => contract.setVote(proposal?.id, selectedOption, userVotingPower),
    true,
    cb
  );
};

export const getOptionLabel = ({ metadata, optionKey, t }) => {
  const metadataLabel = metadata?.voteOptions?.[optionKey];
  return metadataLabel
    ? metadataLabel
    : Number(optionKey) === 0
    ? t('against', { defaultValue: 'Against' })
    : t('option', { optionKey });
};
