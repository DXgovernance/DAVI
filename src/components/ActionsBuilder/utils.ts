import { DecodedAction } from './types';
import { Permission } from './types';
import Web3 from 'web3';
import { NULL_SIGNATURE } from 'utils';
import { SupportedAction } from './types';

const web3 = new Web3();

export const getPermissionArgs = (
  decodedActions: DecodedAction[]
): Permission[] => {
  const permissions = decodedActions.map(decodedAction => {
    // Native transfer of ETH does not have a function
    if (
      decodedAction?.decodedCall?.callType ===
        SupportedAction.NATIVE_TRANSFER ||
      decodedAction?.decodedCall?.callType === SupportedAction.RAW_TRANSACTION
    ) {
      return {
        from: decodedAction?.decodedCall?.from,
        to: decodedAction?.decodedCall?.to,
        callType: decodedAction?.decodedCall?.callType,
        functionSignature: NULL_SIGNATURE,
      };
    }
    const functionInputs = decodedAction?.decodedCall?.function?.inputs?.map(
      input => input.type
    );
    const functionName = decodedAction?.decodedCall?.function?.name;
    const functionSignature = web3.eth.abi.encodeFunctionSignature(
      `${functionName}(${functionInputs.join(',')})`
    );
    return {
      from: decodedAction?.decodedCall?.from,
      to: decodedAction?.decodedCall?.to,
      callType: decodedAction?.decodedCall?.callType,
      functionSignature,
    };
  });
  return permissions;
};
