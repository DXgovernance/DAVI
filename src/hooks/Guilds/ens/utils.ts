// import { isSupportedChainId } from 'components/ActionsBuilder/SupportedActions/UpdateENSContent/utils';
import { MAINNET_ID } from 'utils';

export const isAvailableOnENS = (chainId: number) => {
  const ensNetworks = [MAINNET_ID];
  return ensNetworks.includes(chainId);
};
