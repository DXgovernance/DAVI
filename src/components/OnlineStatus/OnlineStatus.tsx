import React from 'react';
import useOnlineStatus from 'hooks/Guilds/useOnlineStatus';
import { OnlineStatusWrapper, Title, SubTitle } from './OnlineStatus.styled';
import { RiWifiOffLine } from 'react-icons/ri';
import { Flex } from 'components/primitives/Layout/Flex';

export const OnlineStatus: React.FunctionComponent<{
  children: React.ReactElement;
}> = ({ children }) => {
  const { isOnline } = useOnlineStatus();

  return isOnline ? (
    children
  ) : (
    <OnlineStatusWrapper data-testid="offline-wrapper">
      <Flex>
        <RiWifiOffLine size={60} />
        <Title>You are not Online</Title>
        <SubTitle> Please check your internet connection</SubTitle>
      </Flex>
    </OnlineStatusWrapper>
  );
};
