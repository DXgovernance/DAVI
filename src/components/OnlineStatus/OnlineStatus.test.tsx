import { OnlineStatus } from './OnlineStatus';
import useOnlineStatus from 'hooks/Guilds/useOnlineStatus';
import { render } from 'utils/tests';

jest.mock('hooks/Guilds/useOnlineStatus');

describe('OnlineStatus', () => {
  it('Should match snapshot', () => {
    (useOnlineStatus as any).mockImplementation(() => ({ isOnline: true }));
    const { container } = render(
      <OnlineStatus>
        <h1>TEST</h1>
      </OnlineStatus>
    );
    expect(container).toMatchSnapshot();
  });
  it('Should render children', () => {
    (useOnlineStatus as any).mockImplementation(() => ({ isOnline: true }));
    const { getByText } = render(
      <OnlineStatus>
        <h1>TEST</h1>
      </OnlineStatus>
    );
    expect(getByText('TEST')).toBeInTheDocument();
  });

  it('Should render offline box', () => {
    (useOnlineStatus as any).mockImplementation(() => ({ isOnline: false }));
    const { getByTestId } = render(
      <OnlineStatus>
        <h1>TEST</h1>
      </OnlineStatus>
    );
    expect(getByTestId('offline-wrapper')).toBeInTheDocument();
  });
});
