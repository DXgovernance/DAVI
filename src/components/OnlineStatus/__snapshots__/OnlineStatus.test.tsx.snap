// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnlineStatus Should match snapshot 1`] = `
<div>
   
  <h1>
    TEST
  </h1>
   
</div>
`;
